app/core/Engine/Engine.test.ts

```diff
@@ -85,6 +85,142 @@ describe('Engine', () => {
     );
   });
 
+  it('setSelectedAccount successfully updates selected account when address exists', () => {
+    const engine = Engine.init(backgroundState);
+    const validAddress = '0x9DeE4BF1dE9E3b930E511Db5cEBEbC8d6F855Db0';
+    const mockAccount = {
+      id: 'account-1',
+      address: validAddress,
+      type: 'eip155:eoa' as const,
+      options: {},
+      metadata: {
+        name: 'Test Account',
+        importTime: Date.now(),
+        keyring: {
+          type: 'HD Key Tree',
+        },
+      },
+      scopes: [],
+      methods: [],
+    };
+
+    const getAccountByAddressSpy = jest
+      .spyOn(engine.context.AccountsController, 'getAccountByAddress')
+      .mockReturnValue(mockAccount);
+
+    const setSelectedAccountSpy = jest
+      .spyOn(engine.context.AccountsController, 'setSelectedAccount')
+      .mockImplementation();
+
+    const setSelectedAddressSpy = jest
+      .spyOn(engine.context.PreferencesController, 'setSelectedAddress')
+      .mockImplementation();
+
+    engine.setSelectedAccount(validAddress);
+
+    expect(getAccountByAddressSpy).toHaveBeenCalledWith(validAddress);
+    expect(setSelectedAccountSpy).toHaveBeenCalledWith(mockAccount.id);
+    expect(setSelectedAddressSpy).toHaveBeenCalledWith(validAddress);
+
+    getAccountByAddressSpy.mockRestore();
+    setSelectedAccountSpy.mockRestore();
+    setSelectedAddressSpy.mockRestore();
```

Comment:
Could restore all mocks at once maybe?

```suggestion
    jest.restoreAllMocks();
```

Reply:
Also, another question, shouldn't we be using `afterEach` to restore all mocks everytime? If one of those tests fails, the mocks might "bleed out" on other tests, which can waterfall and makes them fail too (if some mocks state are "incorrect")?

Maybe that's the pattern we use on mobile, but just wanted to raise that point, just in case. 😄

Reply:
yeah we should. its much cleaner, I was just following the oudated pattern set in this test already but an afterEach is better.